app.py

```diff
@@ -20,15 +22,28 @@ def get_model(self, num_labels: int):
         )
         return model, tokenizer
 
-    def get_dataset_name(self):
-        return "YelpReviewFull", 5
+    def get_dataset(self):
+        data_root_path = os.path.expanduser("~/.cache/torchtext")
+        train_dset = torchtext.datasets.YelpReviewFull(
+            root=data_root_path, split="train"
+        )
+        val_dset = torchtext.datasets.YelpReviewFull(root=data_root_path, split="test")
+        num_labels = 5
+        return train_dset, val_dset, num_labels
+
+    def get_trainer_settings(self):
+        return dict(strategy="deepspeed_stage_3_offload", precision=16)
+
+    def finetune(self):
+        train_dset, val_dset, num_labels = self.get_dataset()
+        module, tokenizer = self.get_model(num_labels)
+        pl_module = TextClassification(model=module, tokenizer=tokenizer)
+        datamodule = TextClassificationData(
+            train_dataset=train_dset, val_dataset=val_dset, tokenizer=tokenizer
+        )
+        trainer = L.Trainer(**self.get_trainer_settings())
```

Comment:
If we're basically returning the dict with the strategy and precision here, do we need to expose this to the end user? If the user is going to change the strategy from deepspeed to something else here, would this component still work as expected? If we haven't tested this with confidence, we should not expose `get_trainer_settings()` to the user in the component.

Reply:
get_trainer_settings is just there to organize code. The user can delete it if they want and directly modify finetune(). 
The strategy needs to be set somewhere. We can't automatically choose it at the moment. It depends on the model and tuning for throughput/memory.

Reply:
So depending on the set strategy value it will still work. But the user choice has to be appropriate for the chosen model. You cannot give a single device strategy when you have a model too big for a single device for example